File: leetcode/twoSum/TwoSum.java
package twoSum;

import java.util.HashMap;
import java.util.Map;

// https://leetcode.com/problems/two-sum/
public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> m = new HashMap<>();  // num -> idx

        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (m.containsKey(diff)) {
                return new int[]{m.get(diff), i};
            }
            m.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: leetcode/twoSum/TwoSum.py
class TwoSum:
    def twoSum(self, nums: 'List[int]', target: int) -> 'List[int]':
        d = {}
        for i in range(0, len(nums)):
            diff = target - nums[i]
            if diff in d:
                return [i, d[diff]]
            d[nums[i]] = i
